src/errors.rs

```diff
@@ -0,0 +1,39 @@
+use std::fmt::{Display, Formatter};
+
+#[derive(Debug)]
+pub enum UrlParseError {
+    NoPath,
+    NotHttps,
+    Parser(url::ParseError),
+}
```

Comment:
Do you really need this, or does `url`'s `ParseError` cover everything? It seems to me that it does: https://docs.rs/url/latest/url/enum.ParseError.html#variants